File: GFG/Strings/checksubseq.py
def subseq(s1,s2):
    n1=len(s1)
    n2=len(s2)
    i=0
    j=0
    while(i<n1 and j<n2):
        if(s1[i]==s2[j]):
            i+=1
            j+=1
        i+=1
    return j==n2

q1="ABCDEFG"
q2="ADG"
print(subseq(q1,q2))

File: GFG/Strings/checksubseq.cpp
#include<iostream>
#include<string>
using namespace std;

bool subseq(string s1,string s2){
    int n1=s1.size();
    int n2=s2.size();
    int i=0,j=0;
    while(i<n1 && j<n2){
        if(s1[i]==s2[j]){
            i++;j++;
        }
        i++;
    }
    if(j!=n2){
        return false;
    }
    else{return true;}
}

int main(){
    string q1="ABCDEFG";
    string q2="AGD";
    cout<<subseq(q1,q2);
    return 0;
}